google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/MavenBuildDeploymentSourceType.java

```diff
@@ -83,9 +83,12 @@ protected BeforeRunTask createBuildTask(@NotNull Module module) {
   @Override
   public MavenBuildDeploymentSource load(@NotNull Element tag, @NotNull Project project) {
     final String moduleName = tag.getAttributeValue(NAME_ATTRIBUTE);
+    String environmentName = tag.getAttributeValue(
+        AppEngineDeploymentConfiguration.ENVIRONMENT_ATTRIBUTE);
+    AppEngineEnvironment environment = AppEngineEnvironment.valueOf(environmentName);
```

Comment:
I'm struggling to recall the details but I do know that this crashing due to a missing environment did not occur previously even though we were not serializing the environment. I wonder if that crashing is indicative of a different problem.

I remember there was a reason though why the environment was not being serialized. I'll update once I can recall.